screeps-game-api/src/objects/mod.rs

```diff
@@ -225,12 +225,6 @@ pub unsafe trait RoomObjectProperties:
 /// The reference returned by `AsRef<Reference>::as_ref` must reference a
 /// JavaScript object extending the `Structure` class.
 pub unsafe trait StructureProperties: RoomObjectProperties {
-    fn hits(&self) -> i32 {
-        js_unwrap!(@{self.as_ref()}.hits)
-    }
-    fn hits_max(&self) -> i32 {
-        js_unwrap!(@{self.as_ref()}.hitsMax)
-    }
```

Comment:
It's actually something I didn't think about since `Attackable` is implemented for all `StructureProperties` right now (I checked!), including `StructureController`.

Also, the situation is even worse than what you mentioned: calling `hits` on a controller returns `undefined`. This leads me to believe that the right solution might be to leave `hits` on `Attackable` and instead remove `Attackable` from `StructureController`. (To be fair, the game will let you attack it, but it's invincible.)

Your thoughts?

Reply:
Oh- I had not realized `hits` on `StructureController` was undefined! It's there in the documentation and I assumed that was accurate. Thank you for the thorough investigation.

Being able to attack it in screeps makes sense, I guess. It's kind of loosely-constrained and I can see it not hurting them to include it. Wish they'd returned `0` from `hits` and `hitsMax`, but not everything can be perfect.

I think since we have `StructureProperties` and `Attackable` as separate types, we might as well make them as exact as we can. Let's go ahead and remove `Attackable` along with `hits` / `hitsMax` from `StructureController`?

There hasn't been any formal decision on whether we should try to make the Rust bindings more exact than the JS functionality, but it seems alright to me. Not sure if this is a slippery slope, but either way we can always change it later.

---

Since `StructureController` doesn't have `hits` and everything attackable does, keeping those properties on `Attackable` is alright with me.

Are you good with removing the `Attackable` implementation on `StructureController` and leaving the rest of this as it is?

Reply:
Until I get pull access on the repo, it's entirely your decision. 

Now if you want my opinion, I think we should correct the mistakes where possible as those are made glaring due to the type system. We already deviate from the standard API in many ways because we cannot have the type of function parameter representation they have in JS without sacrificing clarity. Same with using enums instead of constants.

Personnaly, I prefer clear and accurate over exact replica. Especially if you think a user of these rust bindings would have to correct type inaccuracies in their own code instead.

Reply:
Alright, glad to hear the input. Clear and accurate sounds good to me.